### ISSUE TITLE
getProof/getHexProof should honour the `hashLeaves` setting

### ISSUE THREAD

**Issue by axic** [miguelmota/merkletreejs#41]
Since it is possible to set up the tree such that it will perform hashing:
```
 const merkleTree = new MerkleTree(elements, keccak256, { hashLeaves: true });
```

I think it would make sense in that case for `getHexProof` to also perform hashing on the leaf.

Was this ever considered?

**Comment by miguelmota** [comment]
I agree that it would make sense to take account the `hashLeaves` option (this was an oversight on my end) but it wouldn't be ideal to break existing behavior so I'll take this into account for the next major release. We could however make an optional argument for hashing options
